### DISCUSSION TITLE
When i use Del(keys...) to delete multiple keys, none of the keys are deleted

### DISCUSSION THREAD

**Discussion by Aaraly** [redis/go-redis#2285] [Q&A]
I want to delete multiple keys by using Del(keys...), because i don't want to delete all these keys one by one, i think this way will reduce program efficiency.
And the finally result is none of the keys are deleted. Is there a problem with my usage?

This is my function to delete keys:

```go
func (db *DbClient) Del(key ...string) error {
	var result *redis.IntCmd
	result = db.rdb.Del(db.ctx, key...)
	_, err := result.Result()
	return err
}
```
Thank you!!

**Comment by vmihailenco** [comment]
The code you provided looks fine. Most like the problem is somewhere else.

**Reply by Aaraly** [reply]
Thanks for your reply. I used this method in cluster mode before, and I found that the key get by using Keys function is not in the currently connected node, is in the other node, so I guest maybe it affects Del(keys...) function. Then I try to use single mode to connect a fixed master node to delete multiple keys, but the deletion still failed.

**Reply by vmihailenco** [reply]
Makes sense. We probably should write a test for a scenario like this.
